### DISCUSSION TITLE
Unauthorized troubleshooting multi cluster

### DISCUSSION THREAD

**Discussion by wolga777** [argoproj/argo-cd#12129] [General]
Good Day everyone,

My problem:
 I'm trying to add a cluster with 
`argocd cluster add <name> --namespace <name> --system-namespace <name> --service-account <name> `. Unfortunately I get the error message Unauthorized. I don't know exactly how to solve this problem/troubleshoot/debug? Are there any logs that show me what exactly ArgoCD was trying to do? If I manually connect to the Kubernetes API with curl and the token, it works. How would you solve this problem?

**Comment by wolga777** [comment]
E.g. I find the parameters --loglevel debug in the argocd cli, but that is not helping me

**Comment by reginaelyse** [comment]
I'm not too sure about your issue because I don't know the full error, but can you try checking what permissions your serviceaccount has? And maybe running `kubectl logs -n argocd -l app.kubernetes.io/name=argocd-server` might give you more logs with some information

**Reply by wolga777** [reply]
Thanks for your answer. I fixed it with `argocd admin cluster kubeconfig https://<api-server-url> /tmp/kubeconfig --namespace argocd` described here->https://argo-cd.readthedocs.io/en/stable/operator-manual/troubleshooting/ 
There I realized I had to change the Token.
